java/pom.xml

```diff
@@ -887,6 +887,99 @@
       </reporting>
     </profile>
 
+    <profile>
+      <id>generate-jnicpp-dylib_so</id>
+      <properties>
+        <ARROW_DATASET>OFF</ARROW_DATASET>
+        <ARROW_GANDIVA>OFF</ARROW_GANDIVA>
+        <ARROW_ORC>OFF</ARROW_ORC>
+      </properties>
+      <build>
+        <plugins>
+          <plugin>
+            <groupId>org.codehaus.mojo</groupId>
+            <artifactId>exec-maven-plugin</artifactId>
+            <version>3.1.0</version>
+            <executions>
+              <execution>
+                <id>jnicppdir</id>
+                <phase>generate-resources</phase>
+                <goals>
+                  <goal>exec</goal>
+                </goals>
+                <configuration>
+                  <executable>mkdir</executable>
+                  <commandlineArgs>-p java-dist java-native-cpp</commandlineArgs>
+                  <workingDirectory>../</workingDirectory>
+                </configuration>
+              </execution>
+              <execution>
+                <id>jnicppdefine</id>
+                <phase>generate-resources</phase>
+                <goals>
+                  <goal>exec</goal>
+                </goals>
+                <configuration>
+                  <executable>cmake</executable>
+                  <commandlineArgs>
+                    -DARROW_BOOST_USE_SHARED=OFF
```

Comment:
We can use `ARROW_DEPENDENCY_USE_SHARED=OFF` instead of multiple `ARROW_*_USE_SHARED=OFF`s.

Reply:
Changed